src/configure/utilities/InputUxDescriptor.ts

```diff
@@ -0,0 +1,130 @@
+import { InputBoxOptions, QuickPickItem } from 'vscode';
+import { IAzureQuickPickOptions } from 'vscode-azureextensionui';
+import { telemetryHelper } from '../helper/telemetryHelper';
+import { ExtendedInputDescriptor, InputMode } from "../model/Contracts";
+import { AzureSession, ControlType, extensionVariables } from '../model/models';
+import { Messages } from '../resources/messages';
+import { TelemetryKeys } from '../resources/telemetryKeys';
+import { DataSourceExpression } from './DataSourceExpression';
+
+export class InputUxDescriptor {
+    public dataSource: DataSourceExpression;
+    public dataSourceUxInputs: Array<InputUxDescriptor>;
+    public dataSourceInputs: Map<string, any>;
+    private input: ExtendedInputDescriptor;
+    private value: any;
+    private controlType: ControlType;
+    private visible: boolean;
+
+    constructor(input: ExtendedInputDescriptor, value: any, controlType: ControlType) {
+        this.input = input;
+        this.value = value;
+        this.visible = true;
+        this.controlType = controlType;
+        this.dataSourceUxInputs = [];
+        this.dataSourceInputs = new Map<string, any>();
+    }
+
+    public getValue(): any {
+        return this.value;
+    }
+
+    public setValue(defaultValue: string) {
+        this.value = defaultValue;
+    }
+
+    public getInputUxDescriptorId(): string {
+        return this.input.id;
+    }
+
+    public getInputGroupId(): string {
+        return this.input.groupId;
+    }
+
+    public getInputMode(): InputMode {
+        return this.input.inputMode;
+    }
+
+    public getParameterValue(): any {
+        return this.value;
+    }
+
+    public getVisibleRule(): string {
+        return this.input.visibleRule;
+    }
+
+    public isVisible(): boolean {
+        return this.visible;
+    }
+
+    public setInputVisibility(value: boolean): void {
+        this.visible = value;
+    }
+    
+    public async setInputUxDescriptorValue(azureSession: AzureSession): Promise<any> {
+        if(!this.isVisible()){
+            return;
+        }
+        if (!!this.dataSource) {
+            var inputs = this._getDataSourceInputs();
```

Comment:
this variable name is literally used at so many places that it does not have a clear meaning. Please user a different name

Reply:
changed to dependentInputs, fixed at other places as well.